packages/rulesets/src/native/legacyRules/Rpaas_ResourceProvisioningState.ts

```diff
@@ -22,14 +22,16 @@ rules.push({
     const allResources = utils.getAllResourceNames()
     for (const resource of allResources) {
       const model = utils.getResourceByName(resource)
+      const azureResource = swaggerUtil?.getProperty(model!, "x-ms-azure-resource")
```

Comment:
I don't think it is right to read "x-ms-azure-resource" as property here. So an existing test is failing. I'll need some help on figuring this part out so that we can run this provisioningState validation only if it is an ARM resource.

Reply:
We have a strategy to detect if a model indicates an arm resource, see https://github.com/Azure/azure-openapi-validator/blob/main/docs/resources/resource-model-validation.md, so we don't need to check the 'x-ms-azure-resource' again here, if you encounter any false alert, please comment and I will take a look